src/_gettsim/transfers/rente.py

```diff
@@ -829,15 +1042,261 @@ def _ges_rente_altersgrenze_vorzeitig(  # noqa: PLR0913
+) -> float:
+    """Early retirement age of pension for unemployed.
+
+    Parameters
+    ----------
+    _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss
+        See :func:`_ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss`.
+
+    Returns
+    -------
+    Early retirement age for unemployed.
+    """
+
+    return _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss
+
+
+@dates_active(
+    start="1996-07-29", end="1996-09-26", change_name="ges_rente_arbeitsl_vorzeitig"
+)
+def ges_rente_arbeitsl_vorzeitig_mit_vertrauenss_1996(
+    ges_rente_params: dict,
+    vertra_arbeitsl_1997: bool,
+    _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss: float,
+) -> float:
+    """Early retirement age of pension for unemployed.
+
+    Includes Vertrauensschutz rules implemented from July to September 1996.
+
+    Parameters
+    ----------
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    geburtsmonat
+        See basic input variable :ref:`geburtsmonat <geburtsmonat>`.
+    vertra_arbeitsl_2006
+        See basic input variable :ref:`vertra_arbeitsl_2006
+        <vertra_arbeitsl_2006>`.
+    _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss
+        See :func:`_ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss`.
+
+    Returns
+    -------
+    Early retirement age for unemployed.
+    """
+
+    if vertra_arbeitsl_1997:
+        arbeitsl_vorzeitig = ges_rente_params[
+            "altersgrenze_arbeitsl_abschlag_vertrauensschutz"
+        ]
+    else:
+        arbeitsl_vorzeitig = _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss
+
+    return arbeitsl_vorzeitig
+
+
+@dates_active(
+    start="1996-09-27", end="2004-07-25", change_name="ges_rente_arbeitsl_vorzeitig"
+)
+def _ges_rente_arbeitsl_vorzeitig_ohne_staffelung_nach_1997(
+    ges_rente_params: dict,
+    geburtsjahr: int,  # noqa: ARG001
+) -> float:
+    """Early retirement age of pension for unemployed.
+
+    Early retirement age does not depend on birth year and month.
+
+    ToDo: Remove fake dependency (geburtsjahr).
+
+    Parameters
+    ----------
+    ges_rente_params
+        See params documentation
+        :ref:`ges_rente_params <ges_rente_params>`.
+
+    Returns
+    -------
+    Early retirement age for unemployed.
+
+    """
+
+    return ges_rente_params["altersgrenze_arbeitsl_vorzeitig"]
+
+
+@dates_active(
+    start="2004-07-26", end="2017-12-31", change_name="ges_rente_arbeitsl_vorzeitig"
+)
+def ges_rente_arbeitsl_vorzeitig_mit_vertrauenss_ab_2006(
+    ges_rente_params: dict,
+    vertra_arbeitsl_2006: bool,
+    _ges_rente_arbeitsl_vorzeitig_ohne_vertrauenss: float,
+) -> float:
+    """Early retirement age of pension for unemployed.
+
+    Includes Vertrauensschutz rules implemented in 2006. Policy becomes inactive in 2018
+    because then all potential beneficiaries have reached the normal retirement age.
+
+    Parameters
```

Comment:
Docstring paramets do not match parameters.

Feel free to ignore for the moment. But we should really enable the "D" codes soon in `pyproject.toml` (maybe excluding some sub-codes), then it will be flagged automatically.